imap_processing/utils.py

```diff
@@ -101,17 +101,22 @@ def convert_raw_to_eu(dataset: xr.Dataset, conversion_table_path, packet_name):
     return dataset
 
 
-def create_dataset(packets: list[Packet], met_name="shcoarse") -> xr.Dataset:
+def create_dataset(
```

Comment:
Ooooo. I didn't know about this function. Love that it is being made more generic. If I am not mistaken it can be used for any product that is generated directly from the packet storing everything in one-dimensional, 12-bit CDF variables? Some details about usage restrictions should be added to the docstring.

I wonder if functionality could be added to allow for custom unpacking of certain keys in the input packets as well as CDF variables that are not type "I12"?

Reply:
I see now that I was misinterpreting the `format="I12"` and now understand that this is to display an integer with a field width of 12. Ignore that part of my comment.

Reply:
Nice improvements to this function!

Reply:
@subagonsouth let me know if you like clarification on this.

yes. I was trying to make this function more generic as I go along.